src/deluge/playback/playback_handler.cpp

```diff
@@ -1883,46 +1884,90 @@ void PlaybackHandler::tempoEncoderAction(int8_t offset, bool encoderButtonPresse
 	}
 
 	// Otherwise, change tempo
-	else {
-		if (!(playbackState & PLAYBACK_CLOCK_EXTERNAL_ACTIVE)) {
-
-			if (Buttons::isButtonPressed(hid::button::TEMPO_ENC)) {
-				// Fine tempo adjustment
-				uint32_t tempoBPM = calculateBPM(currentSong->getTimePerTimerTickFloat()) + 0.5;
-				tempoBPM += offset;
-				if (tempoBPM > 0) {
-					currentSong->setBPM(tempoBPM, true);
-					displayTempoBPM(tempoBPM);
+		else {
+
+			if (runtimeFeatureSettings.get(RuntimeFeatureSettingType::FineTempoKnob)
+		    		== RuntimeFeatureStateToggle::On) {
+
+				if (!(playbackState & PLAYBACK_CLOCK_EXTERNAL_ACTIVE)) {
+
+					if (Buttons::isButtonPressed(hid::button::TEMPO_ENC)) {
```

Comment:
Compare this to the feature toggle instead of nesting if/else

E.g. if tempo_button_pressed == tempo_press_coarse_enabled {}

Reply:
Thanks for the tip... I'm learning as I go...